File: src/ERPWeb.Entity/BasicInfo/GrtWastePercent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ERPWeb.Entity.MacEntity
{
    public class GrtWastePercent
    {
        public int ID { get; set; }
        /// <summary>
        /// 机种和线别
        /// </summary>
        public string ModelLine { get; set; }
        /// <summary>
        /// 班别
        /// </summary>
        public string Shift { get; set; }
        /// <summary>
        /// 已投放托盘数
        /// </summary>
        public int TrayInputNum { get; set; }
        /// <summary>
        /// 已投放GRT总数
        /// </summary>
        public int GRTTotalNum { get; set; }
        /// <summary>
        /// 已消耗的GRT总数
        /// </summary>
        public int GRTShiftNum { get; set; }
        /// <summary>
        /// 浪费掉的GRT(抛料数)
        /// </summary>
        public int GRTWasteNum { get; set; }
        /// <summary>
        /// 浪费掉的GRT比例(抛料率)
        /// </summary>
        public string GRTWastePercent { get; set; }
        /// <summary>
        /// 查询时间
        /// </summary>
        public string CheckDate { get; set; }
    }
}

File: src/ERPWeb.Web/Scripts/Machine/GrtWastePercent.js
//窗体加载
$(function () {
    initGRTTable();
    initGRTClick();
    $('#searchGRTDate').datebox({
        onSelect: function (date)
        {
            $('#searchGRTDate').textbox('setValue', date.getFullYear() +'-' +(date.getMonth() + 1) +'-' +date.getDate());
        }
    });
});



//初始化表格
function initGRTTable() {
    var modelLine = $('#searchGRTInputLine').textbox('getValue');
    var shift = $('#searchGRTInputShift').textbox('getValue');
    var checkDate = $('#searchGRTDate').textbox('getValue');
    $('#GRTTable').datagrid({
        //分页
        fitColumns: true,
        singleSelect: true,
        url: '../../Machine/GetGrtWastePercent',
        queryParams: { ModelLine: modelLine, Shift: shift, CheckDate: checkDate },
        pagination: true,
        rownumbers: true,
        pagePosition: 'bottom',
        pageSize: 10,
        pageNumber: 1,
        pageList: [2, 10, 15, 20, 30, 50],
        toolbar: '#searchGRTToolBar',
        fitColumns: true,
        //数据格式
        columns: [[
            { title: '机种与线别', width: 150, field: 'ModelLine', },
            { title: '班别', width: 150, field: 'Shift', },
            { title: '已投入托盘', width: 150, field: 'TrayInputNum', },
            { title: '已投入GRT总数', width: 150, field: 'GRTTotalNum', },
            { title: '已消耗GRT总数', width: 150, field: 'GRTShiftNum', },
            { title: '抛料数', width: 125, field: 'GRTWasteNum' },
            { title: '抛料率', width: 125, field: 'GRTWastePercent' },
            { title: '查询时间', width: 200, field: 'CheckDate' },
        ]],
    });
}

function initGRTClick() {
    //点击Search按钮
    $('#btnGRTSearch').click(function () {
        reloadGRTTable();//重新加载表格
    });
}

function reloadGRTTable() {
    $('#GRTTable').datagrid('load', {
        modelLine: $('#searchGRTInputLine').textbox('getValue'),
        shift: $('#searchGRTInputShift').textbox('getValue'),
        checkDate: $('#searchGRTDate').textbox('getValue'),
    });
}